### PR TITLE
feat(propfind): add audio and location props

### PR DESCRIPTION
**Pull request by dschmidt**
Return `oc:audio` and `oc:location` props for allprops requests or when explicitly requested.
These are analogue to `audio` and `location` facets defined in LibreGraph API.

Camelcase seems off, should we lowercase or dasherize the prop names? For eg albumArtist

Audio data is already indexed in oCIS master, location data depends on https://github.com/owncloud/ocis/pull/7886
 
```xml
<d:multistatus xmlns:s="http://sabredav.org/ns" xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/spaces/storage-users-1$some-admin-user-id-0000-000000000000/Foo/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>storage-users-1$some-admin-user-id-0000-000000000000!746e171b-7143-4461-b1c0-f119cb57a3d1</oc:id>
        <oc:fileid>storage-users-1$some-admin-user-id-0000-000000000000!746e171b-7143-4461-b1c0-f119cb57a3d1</oc:fileid>
        <oc:spaceid>some-admin-user-id-0000-000000000000</oc:spaceid>
        <oc:file-parent>storage-users-1$some-admin-user-id-0000-000000000000!some-admin-user-id-0000-000000000000</oc:file-parent>
        <oc:name>Foo</oc:name>
        <d:getetag>&quot;2c7f03b3e0a1efc87ad6da4717ffb9ba&quot;</d:getetag>
        <oc:permissions>RDNVCKZP</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <oc:size>7196356</oc:size>
        <d:getlastmodified>Tue, 05 Dec 2023 16:54:00 GMT</d:getlastmodified>
        <oc:tags/>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
    <d:propstat>
      <d:prop>
        <oc:audio/>
        <oc:location/>
      </d:prop>
      <d:status>HTTP/1.1 404 Not Found</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/dav/spaces/storage-users-1$some-admin-user-id-0000-000000000000/Foo/A_IMG_20231006_185428.jpg</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>storage-users-1$some-admin-user-id-0000-000000000000!e3afec67-5e82-4c7b-884f-2f798029cee4</oc:id>
        <oc:fileid>storage-users-1$some-admin-user-id-0000-000000000000!e3afec67-5e82-4c7b-884f-2f798029cee4</oc:fileid>
        <oc:spaceid>some-admin-user-id-0000-000000000000</oc:spaceid>
        <oc:file-parent>storage-users-1$some-admin-user-id-0000-000000000000!746e171b-7143-4461-b1c0-f119cb57a3d1</oc:file-parent>
        <oc:name>A_IMG_20231006_185428.jpg</oc:name>
        <d:getetag>&quot;26c5b396fa752e90f271102690f61fd5&quot;</d:getetag>
        <oc:permissions>RDNVWZP</oc:permissions>
        <d:resourcetype/>
        <d:getcontentlength>1347813</d:getcontentlength>
        <d:getcontenttype>image/jpeg</d:getcontenttype>
        <d:getlastmodified>Tue, 05 Dec 2023 15:59:36 GMT</d:getlastmodified>
        <oc:checksums>
          <oc:checksum>SHA1:da95c8511b6e4bbc9f580457d8b36ca9599b1862 MD5:d365cb975fa598b49289a348d32247d5 ADLER32:c586519c</oc:checksum>
        </oc:checksums>
        <oc:tags/>
        <oc:location>
          <oc:latitude>49.801781</oc:latitude>
          <oc:longitude>9.932846</oc:longitude>
        </oc:location>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
    <d:propstat>
      <d:prop>
        <oc:audio/>
      </d:prop>
      <d:status>HTTP/1.1 404 Not Found</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/dav/spaces/storage-users-1$some-admin-user-id-0000-000000000000/Foo/01%20-%20Sucker.mp3</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>storage-users-1$some-admin-user-id-0000-000000000000!6f3f2980-6dd7-4b4a-b01c-722bf3d53ec0</oc:id>
        <oc:fileid>storage-users-1$some-admin-user-id-0000-000000000000!6f3f2980-6dd7-4b4a-b01c-722bf3d53ec0</oc:fileid>
        <oc:spaceid>some-admin-user-id-0000-000000000000</oc:spaceid>
        <oc:file-parent>storage-users-1$some-admin-user-id-0000-000000000000!746e171b-7143-4461-b1c0-f119cb57a3d1</oc:file-parent>
        <oc:name>01 - Sucker.mp3</oc:name>
        <d:getetag>&quot;5897769a2eaec9573c6f0919464e1796&quot;</d:getetag>
        <oc:permissions>RDNVWZP</oc:permissions>
        <d:resourcetype/>
        <d:getcontentlength>5848543</d:getcontentlength>
        <d:getcontenttype>audio/mpeg</d:getcontenttype>
        <d:getlastmodified>Fri, 13 Oct 2023 16:00:21 GMT</d:getlastmodified>
        <oc:checksums>
          <oc:checksum>SHA1:28b306e048d26df4c53e66263959a5c0dc267204 MD5:4aaad82c633c17db5f94fd38c1c89082 ADLER32:072717cd</oc:checksum>
        </oc:checksums>
        <oc:tags/>
        <oc:audio>
          <oc:album>Kiss Of Death</oc:album>
          <oc:albumArtist>Foobar</oc:albumArtist>
          <oc:artist>Motörhead</oc:artist>
          <oc:disc>4</oc:disc>
          <oc:genre/>
          <oc:title>Sucker</oc:title>
          <oc:year>2006</oc:year>
        </oc:audio>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
    <d:propstat>
      <d:prop>
        <oc:location/>
      </d:prop>
      <d:status>HTTP/1.1 404 Not Found</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

### CODE REVIEW

**Review by kobergj — COMMENTED** [review]
Looks good :+1: But let's go with all lower prop names, most of the existing follow that convention.

**Comment by dschmidt** [comment]
> ```
> file-parent
> ```

`file-parent` has a dash - no idea and no strong feelings on what is better or "more consistent"

**Comment by kobergj** [comment]
Yes. But I think `file-parent` is the only outlier

**Comment by dschmidt** [comment]
Updated

File: internal/http/services/owncloud/ocdav/propfind/propfind.go [hunk 1]
```diff
@@ -64,6 +64,32 @@ const (
...
+		"discCount",
+		"duration",
+		"genre",
+		"hasDrm",
+		"isVariableBitrate",
+		"title",
+		"track",
+		"trackCount",
+		"year",
+	}
```
Comment (kobergj) [inline internal/http/services/owncloud/ocdav/propfind/propfind.go:85]: Can we have them all lower here too? We can still keep the `strings.ToLower` but having them camel case here will lead to confusion. If this is not possible, could you make a comment stating that they will be `ToLower`d later?

**Review by kobergj — CHANGES_REQUESTED** [review]

File: internal/http/services/owncloud/ocdav/propfind/propfind.go (hunk 1 above)
Comment (dschmidt) [reply internal/http/services/owncloud/ocdav/propfind/propfind.go:85]: No, we can't. We used these keys not only for the prop names but also to look up the values in the metadata storage through keys generated with `metadataKeysWithPrefix("libre.graph.audio", audioKeys)...)`
They are camel case just as in the LibreGraph facet.

Comment (kobergj) [reply internal/http/services/owncloud/ocdav/propfind/propfind.go:85]: Ah I see. No need to rename. But please add a comment that they are also used as propnames all lowered. That will make later debugging/refactoring/feature adding easier.

**Review by kobergj — APPROVED** [review]
Cool :eagle:

**Pull request merged**